### ISSUE TITLE
Consider making `DecodeError` and `ValidationError` inherit from ValueError

### ISSUE THREAD

**Issue by vytas7** [jcrist/msgspec#634]
### Description

In many scenarios and libraries one assumes that validation errors inherit from `ValueError`, however, `msgspec`'s exception classes do not. Would it be possible to make [`DecodeError`](https://jcristharif.com/msgspec/api.html#msgspec.DecodeError) and [`ValidationError`](https://jcristharif.com/msgspec/api.html#msgspec.ValidationError) subclasses of `ValueError` somewhere in the hierarchy?

**Comment by jcrist** [comment]
Makes sense to me, happy to make this change. This would be a breaking change of course, but since we're still in `0.*` that shouldn't be unexpected.

**Comment by Siyet** [comment]
Ported to the community fork [msgspec-arise](https://github.com/Siyet/msgspec-arise): [issue #27](https://github.com/Siyet/msgspec-arise/issues/27), [PR #34](https://github.com/Siyet/msgspec-arise/pull/34)
